src/app/navigation/types.ts

```diff
@@ -1,14 +1,28 @@
 import { Location } from 'history';
+import { routes } from '../';
 
 export type State = Location;
 
-export interface Match<Params> {
-  route: Route<Params>;
-  params: Params;
+type RouteParams<R> = R extends Route<infer P> ? P : never;
```

Comment:
I'm having trouble parsing this... oh, it looks like it's something like the following (which doesn't work):

```ts
type RouteParams<_ extends Route<P>> = P
```